File: 2223B/funmock6/10212_Amanda.cpp
/*
*OnlineJudge
*10212 - The Last Non-zero Digit
*Code by: Amanda Lim
*/

#include <bits/stdc++.h>
using namespace std;

int main ()
{
    long long N, M;
    while (cin >> N >> M) {
        long long ans = 1;
        for (long long i = N; i >= N - M + 1; i--) {
            ans *= i;

            while (ans%10 == 0) // to avoid overflow + remove unneeded zeroes
                ans /= 10;
            
            ans %= 1000000000; // to avoid overflow
        }
        cout << ans%10 << endl; // get last nonzero digit
    }
    return 0;
}

File: 2223B/funmock6/10212_Amanda.py
'''
OnlineJudge
10212 - The Last Non-zero Digit
Code by: Amanda Lim
'''

from sys import stdin

# WA
def permutation (A, B):
    ans = 1
    diff = B - A
    diff //= 10
    print(A, B, diff, 8*diff)
    ans *= 8*diff
    
    if (ans == 0):
        ans = 1

    while (A%10 != 0):
        ans *= A
        A += 1
    while (B%10 != 0):
        ans *= B
        B -= 1

    while (ans%10 == 0):
        ans //= 10

    return ans%10

# TLE
def perm (A, B):
    ans = 1
    for i in range(A, B+1):
        ans *= i
        print(ans)
    while (ans%10 == 0):
        ans //= 10

    return ans%10 

for line in stdin:
    inp = line.split(' ')
    N = int(inp[0])
    M = int(inp[1])
    print(permutation(N-M+1, N))
    print(perm(N-M+1,N))
